### ISSUE TITLE
Failing to delete a remote branch isn't error-worthy?

### ISSUE THREAD

**Issue by joshaber** [desktop/desktop#1829]
1. Create a new branch.
1. Go to dotcom.
1. Open a PR from branch
1. Merge PR and delete branch from dotcom.
1. Go back to Desktop.
1. Delete branch.
1. Get an error about how it couldn't delete the remote branch.

It makes sense that the remote branch deletion failed since we just deleted the branch from the Pull Request view. But it's not noteworthy for the user.

Of course the catch is that sometimes it _might_ be noteworthy. Maybe we should fetch + prune before trying to delete the remote branch?

**Comment by terales** [comment]
See a record: https://youtu.be/xCVKG1d93Ow

It's really unclear. Looks like there was an error about removing *local* branch.

Maybe we can add a n indication that the local branch was already successfully deleted and a remote one was already deleted?

**Comment by terales** [comment]
[Found](https://stackoverflow.com/a/30524983/1363799) a more solid solution. We can check if remote branch exists with
```
git ls-remote --heads git@github.com:desktop/desktop.git branch-name
```

If branch exists it would respond with zero and some data,
if there is no such branches it [would respond with zero and empty data](https://git-scm.com/docs/git-ls-remote.html#git-ls-remote---exit-code).

If there is no such branches we shouldn't run [`deleteBranch` command](https://github.com/desktop/desktop/blob/43be7fabe83f42b6e52db33fde8599169c127f06/app/src/lib/git/branch.ts#L49).

Should I close previous pull request and implement this explicit check?

**Comment by shiftkey** [comment]
@terales that seems like a nicer solution than running a full fetch. This can also be tweaked to use the remote name instead of the full URL:

```
git ls-remote --heads origin branch-name
```

I'm happy for you to fold this into #1834 and see how that works.

**Comment by terales** [comment]
@shiftkey I'll update pull request with another solution and remove skipped test in three days. Fell free to assign this issue to me.

**Comment by shiftkey** [comment]
Fixed by #1834
